app/javascript/blacklight/search_context.js

```diff
@@ -54,6 +62,68 @@ const SearchContext = (() => {
   };
 
   Blacklight.doSearchContextBehavior();
+
+  const clientAppliedParams = function(storedSearch) {
+    if (storedSearch) {
+      let backToCatalogParams = new URLSearchParams();
+      for (const [paramName, paramValue] of storedSearch.entries()) {
+        backToCatalogParams.append(paramName, paramValue);
+      }
+      const appliedParams = document.querySelector('#appliedParams')
+      if (appliedParams) {
+        const backToCatalogEle = appliedParams.querySelector('.back-to-catalog')
+        if (backToCatalogEle) {
+          const backToCatalogUrl = new URL(backToCatalogEle.href)
+          backToCatalogEle.href = `${backToCatalogUrl.pathname}?${backToCatalogParams.toString()}`
+          backToCatalogEle.classList.remove('d-none')
+        }
+      }
+    }
+  }
+  const clientPrevNext = function(prevNextPath, storedSearch) {
+    const prevNextUrl = new URL(prevNextPath, window.location)
+    // prevNextUrl should have a counter param, but needs search params
+    let prevNextParams = new URLSearchParams(prevNextUrl.search);
+    if (storedSearch) {
+      for (const [paramName, paramValue] of storedSearch.entries()) {
+        prevNextParams.append(paramName, paramValue);
+      }
+    }
+    const setHrefOrDelete = function(linkEle, url) {
+      if (!linkEle) return;
+      if (url) {
+        linkEle.href = url
+      } else {
+        linkEle.remove()
+      }
+    }
+    const setContent = function(ele, content) {
+      if (!ele) return;
+      ele.innerHTML = content;
+    }
+    fetch(`${prevNextUrl.pathname}?${prevNextParams.toString()}`)
+    .then((response) => response.json())
+    .then(function(responseData) {
+      if (!responseData.prev && !responseData.next) return;
+      document.querySelectorAll('.page-links').forEach(function(pageLinks) {
+        setContent(pageLinks.querySelector('.pagination-counter-raw'), responseData.counterRaw)
+        setContent(pageLinks.querySelector('.pagination-counter-delimited'), responseData.counterDelimited)
+        setContent(pageLinks.querySelector('.pagination-total-raw'), responseData.totalRaw)
+        setContent(pageLinks.querySelector('.pagination-total-delimited'), responseData.totalDelimited)
+        setHrefOrDelete(pageLinks.querySelector("a[rel='prev']"), responseData.prev)
+        setHrefOrDelete(pageLinks.querySelector("a[rel='next']"), responseData.next)
+        pageLinks.classList.remove('d-none')
+      })
+    })
+  }
+  Blacklight.onLoad(function(){
```

Comment:
Is it necessary to use onLoad here or can this function be called by `doSearchContextBehavior`?  I think it's confusing to depend on both approaches.

Reply:
`doSearchContextBehavior` adds an event listener for clicks, which doesn't inspect the DOM (yet). The "page links" behavior in the show view depends on DOM content. I'm trying to refactor.

Reply:
ok, pushed a refactor up. I think that I have to either piggyback on `Blacklight.onLoad` 's event detection, or I would have to reimplement it. I at least relocated the event handler attachment and the onload addition to `doSearchContextBehavior`.